bundles/model/org.eclipse.smarthome.model.rule.runtime/src/org/eclipse/smarthome/model/rule/runtime/internal/engine/RuleEngineImpl.java

```diff
@@ -251,12 +266,18 @@ public void modelChanged(String modelName, org.eclipse.smarthome.model.core.Even
                         && (type == org.eclipse.smarthome.model.core.EventType.ADDED || type == org.eclipse.smarthome.model.core.EventType.MODIFIED)) {
                     triggerManager.addRuleModel(model);
                     // now execute all rules that are meant to trigger at startup
-                    runStartupRules();
+                    scheduleStartupRules();
                 }
             }
         }
     }
 
+    private void scheduleStartupRules() {
+        if (startupJob == null || startupJob.isCancelled() || startupJob.isDone()) {
+            startupJob = scheduler.schedule(startupRunnable, 5, TimeUnit.SECONDS);
```

Comment:
I thought, we could cancel the current scheduled service and schedule it again, but if every 4 seconds a new item is added, the rule is never triggered. So, forget my comment...